### PR TITLE
Lint markdown files on CI

### LINKED ISSUE #268: Lint Documentation MD files on CI
## 📚 Context

The node docs is comprised of several MDfiles, either manually created or automatically generated.

Sometimes a developer might make changes that are not formatted right and such mistakes go unnoticed.

> This is a follow-up to #266 

## ✔️ Solution

Consider enabling markdown at `.golangci.yml`.

If not possible, add a linter action to CI.

Linters below may be considered:

- https://github.com/DavidAnson/markdownlint
- https://github.com/tcort/markdown-link-check

## 📈 Subtasks

- [x] Add MD linter action to CI
- [x] Add corresponding recipe to Makefile
- [x] Disable linter rules on non-documentation files

### PR DESCRIPTION
**Pull request by marcelstanley**
- Add [markdownlint ](https://github.com/DavidAnson/markdownlint-cli2-action) to the CI
- Lint docs
- Fix linter issues
- Disable issues related to https://pkg.go.dev/github.com/spf13/cobra/doc, which would require a custom doc generator to be fixed (see https://github.com/spf13/cobra/blob/bcfcff729ecdeb4072ef6f2a9c0b5e4ca1853206/doc/md_docs.go#L57)
- Add make recipe

Closes #268

### CODE REVIEW

**Review by gligneul — COMMENTED** [review]
Looks great! Could you add an entry to the Makefile as well? Something like `make md-lint`.

**Comment by marcelstanley** [comment]
> Looks great! Could you add an entry to the Makefile as well? Something like `make md-lint`.

Sure, I completely forgot it! 😅

File: Makefile [hunk 1]
```diff
@@ -15,6 +15,13 @@ lint: ## Run the linter
 	@echo "Running the linter"
 	@golangci-lint run
 
+.PHONY: md-lint
+md-lint: ## Lint Markdown docs
+	@echo "Running markdownlint-cli"
+	# Each directory has its own .markdownlint.yaml configuration
```
Comment (gligneul) [inline Makefile:21]:
```suggestion
	@# Each directory has its own .markdownlint.yaml configuration
```

Currently, this is being printed on stdout. I think this was not planned.
```
➜  rollups-node-2 git:(829095fe) make md-lint
Running markdownlint-cli
# Each directory has its own .markdownlint.yaml configuration
```

**Review by gligneul — CHANGES_REQUESTED** [review]

**Review by gligneul — APPROVED** [review]
LGTM

**Pull request merged**